File: Java/Program/RAM.java
public class RAM extends KomponenInternal {
    private int kapasitasGB;
    private String tipe;

    // Constructor
    public RAM() {
        super();
        this.kapasitasGB = 0;
        this.tipe = "";
    }

    public RAM(String merk, String nama, int harga, String jenisSlot, boolean terpasang,
               int kapasitasGB, String tipe) {
        super(merk, nama, harga, jenisSlot, terpasang);
        this.kapasitasGB = kapasitasGB;
        this.tipe = tipe;
    }

    // Setter methods
    public void setKapasitasGB(int kapasitasGB) { this.kapasitasGB = kapasitasGB; }
    public void setTipe(String tipe) { this.tipe = tipe; }

    // Getter methods
    public int getKapasitasGB() { return kapasitasGB; }
    public String getTipe() { return tipe; }

    // Tampilkan Info
    @Override
    public String tampilkanInfo() {
        return super.tampilkanInfo() + " | Kapasitas: " + kapasitasGB + " GB | Tipe: " + tipe;
    }
}

File: Python/Program/RAM.py
from KomponenInternal import KomponenInternal

class RAM(KomponenInternal):
    def __init__(self, merk="", nama="", harga=0, jenis_slot="", terpasang=False, kapasitas_gb=0, tipe=""):
        super().__init__(merk, nama, harga, jenis_slot, terpasang)
        self.kapasitas_gb = kapasitas_gb
        self.tipe = tipe
    
    # Setter
    def set_kapasitas_gb(self, kapasitas_gb):
        self.kapasitas_gb = kapasitas_gb
    
    def set_tipe(self, tipe):
        self.tipe = tipe
    
    # Getter
    def get_kapasitas_gb(self):
        return self.kapasitas_gb
    
    def get_tipe(self):
        return self.tipe
    
    # Tampilkan Info
    def tampilkan_info(self):
        return f"{super().tampilkan_info()} | Kapasitas: {self.kapasitas_gb} GB | Tipe: {self.tipe}"